### PR TITLE
fix(insights): overlay insights charts on top of detail panel

### PR DESCRIPTION
**Pull request by KevinL10**
Overlays the full-screen insights chart on top of the side panel.

image

### CODE REVIEW

File: static/app/components/modals/insightChartModal.tsx [hunk 1]
```diff
@@ -13,7 +13,20 @@ export type InsightChartModalOptions = {
...
-export default function InsightChartModal({Header, title, children}: Props) {
+export default function InsightChartModal({
+  Header,
+  title,
+  children,
+  modalContainerRef,
+}: Props) {
+  useEffect(() => {
+    // display the fullscreen insight charts on top of any sidebar elements
+    // with z-index 10000 or 10001
```
Comment (DominikB2014) [inline static/app/components/modals/insightChartModal.tsx:24]: I'm kind of suprised we have to do this, typically a modal should always be on top, no matter what what. If we have to manually update the zIndex for the modal, i feel as though the sidebar's zIndex is too high?

Comment (KevinL10) [reply static/app/components/modals/insightChartModal.tsx:24]: the modal has z-index 10000 but the sidebar has z-index 10001. I think this is the only place in the product where we possibly display both a modal and a sidebar

Comment (DominikB2014) [reply static/app/components/modals/insightChartModal.tsx:24]: Could we make the sidebar zIndex 9999?

Comment (DominikB2014) [reply static/app/components/modals/insightChartModal.tsx:24]: We would be future proofing ourselves here, there may be more cases for sidebar and modal at some point

**Review by DominikB2014 — APPROVED** [review]
👍

**Pull request merged**